README.md

```diff
@@ -1,21 +1,293 @@
+
+Different open source software is available to the community to manage mesh adaptation: [AMReX](https://amrex-codes.github.io/amrex/) for patch-based AMR, [p4est](https://www.p4est.org/) and [pablo](https://optimad.github.io/PABLO/) for cell-based adaptation.
+
+The strength of samurai is that it allows to implement all the above mentioned mesh adaptation methods from the same data structure. The mesh is represented as intervals and a set algebra allows to efficiently search for subsets among these intervals.
+Samurai also offers a flexible and pleasant interface to easily implement numerical methods.
+
+<details>
+<summary>Table of Contents</summary>
+
+- [Get started](#get-started)
+  - [The advection equation](#the-advection-equation)
+  - [The projection operator](#the-projection-operator)
+  - [There's more](#theres-more)
+- [Features](#features)
+- [Installation](#installation)
+  - [From conda](#from-conda)
+  - [From vcpkg](#from-vcpkg)
+  - [From conan](#from-conan)
+  - [From source](#from-source)
+- [Get help](#get-help)
+- [Project assistance](#project-assistance)
+- [Contributing](#contributing)
+- [License](#license)
+
+</details>
+
+## Get started
+
+In this section, we propose two examples: the first one solves a 2D advection equation with mesh adaptation using multiresolution, the second one shows the use of set algebra on intervals.
+
+### The advection equation
+
+We want to solve the 2D advection equation given by
+
+$$
+\partial_t u + a \cdot \nabla u = 0 \\; \text{in} \\; [0, 1]\times [0, 1]
+$$
+
+with homogeneous Dirichlet boundary conditions and $a = (1, 1)$. The initial solution is given by
+
+$$
+u_0(x, y) = \left\\{
+\begin{align*}
+1 & \\; \text{in} \\; [0.4, 0.6]\times [0.4, 0.6], \\
+0 & \\; \text{elsewhere}.
+\end{align*}
+\right.
+$$
+
+To solve this equation, we use the well known [upwind scheme](https://en.wikipedia.org/wiki/Upwind_scheme).
+
+The following steps describe how to solve this problem with samurai. It is important to note that these steps are generally the same whatever the equations we want to solve.
+
+- Define the configuration of the problem
+
+    ```cpp
+    constexpr size_t dim = 2;
+    using Config = samurai::MRConfig<dim>;
+    using mesh_id_t = typename Config::mesh_id_t;
+
+    std::size_t min_level = 2, max_level = 8;
+    ````
+
+- Create the Cartesian mesh
+
+    ```cpp
+    const samurai::Box<double, dim> box({0., 0.}, {1., 1.});
+    samurai::MRMesh<Config> mesh(box, min_level, max_level);
+    ```
+
+- Create the field on this mesh
+
+    ```cpp
+    auto u = samurai::make_field<double, 1>("u", mesh);
+    samurai::make_bc<samurai::Dirichlet>(u, 0.);
+    ```
+
+- Initialization of this field
+
+    ```cpp
+    samurai::for_each_cell(mesh[mesh_id_t::cells], [&](const auto& cell)
```

Comment:
As it's the default behaviour, maybe we could remove `[mesh_id_t::cells]`...? That would simplify the example by getting rid of all the `mesh_id_t`.